cfgov/unprocessed/css/molecules/inset.less

```diff
@@ -19,32 +26,102 @@
           .o-full-width-text
             .m-full-width-text
             .m-inset
-              .m-quote
-              .m-related-links
-              .m-call-to-action
-            .m-full-width-text
+            .m-inset__bottom-rule
+            .m-inset__left
+            .m-inset__portrait
+            .m-inset__150
+            .m-inset__300
+                .header-slug
+                    .header-slug_inner
+                .m-inset_image-container
+                    div.m-inset_image__cover
+                    div.m-inset_image__decorative
+                    img.m-inset_image
+                .m-inset_text
+                .m-quote
+                .m-related-links
+                .m-call-to-action
+                .m-image-inset
+                .m-full-width-text
   tags:
     - cfgov-molecules
   notes:
     - These are held in a full-width-text-group container to be held with
-      various full-width text blocks,  insets and tables as needed.
+      various full-width text blocks, insets and tables as needed.
 */
 
 .m-inset {
-    margin-top: unit( @grid_gutter-width / @base-font-size-px, em );
-    margin-bottom: unit( @grid_gutter-width / @base-font-size-px, em );
     width: 100%;
     clear: right;
 
-    .respond-to-min(@bp-med-min, {
+    &__bottom-rule {
+        padding-bottom: unit( @grid_gutter-width / @base-font-size-px, em );
+        border-bottom: 1px solid @horizontal-rule;
+    }
+
+    .m-inset_image__decorative {
+        background-size: contain;
+        background-repeat: no-repeat;
+    }
+
+    .m-inset_text {
+        margin-top: unit( 5px / @base-font-size-px, em );
+    }
+
+    .respond-to-min( @bp-med-min, {
         float: right;
         width: unit( 270px / @base-font-size-px, em );
-        margin-left: unit( @grid_gutter-width / @base-font-size-px, em );
-    });
+    } );
+
+    .respond-to-max( @bp-xs-max, {
+
+        .m-inset_image {
+            .u-flexible-container_inner();
+            height: auto;
+            width: auto;
+        }
+
+        .m-inset_image__cover {
+            background-size: cover;
+        }
+
+        .m-inset_image__decorative {
+
+            // Important is necessary to override the inline height value;
+            height: 100% !important;
+            width: 100%;
+        }
+
+        .m-inset_image-container {
+            overflow: hidden;
+            .u-flexible-container();
+        }
+
+        &__portrait {
+            .m-inset_image-container {
+                padding-bottom: 75%;
+            }
+        }
+    } );
+
+    .respond-to-min( @bp-sm-min, {
+        &__150 {
+            width: 150px;
+        }
+
+        &__300 {
+            width: 300px;
+        }
+
+        &__image {
+            float: right;
```

Comment:
Ah ok. I agree it'd be nice to have these unified a bit so it's clear what to expect instead of thinking about the type of inset it is.